benchmark.go

```diff
@@ -41,6 +42,40 @@ var (
 	errNegativeMaxReqs  = errors.New("max requests cannot be negative")
 )
 
+// Struct to hold values of benchmark parameters
```

Comment:
nits: Comments for exposed structs normally starts with the struct name, e.g. 
```
BenchmarkParameters holds values....
```

same for other structs you defined